### PR TITLE
Support DeepSeekV3-style block FP8 quantization

### PR DESCRIPTION
**Pull request by mgoin**
Fixes https://github.com/vllm-project/llm-compressor/issues/1475
Blocked on CT support in https://github.com/neuralmagic/compressed-tensors/pull/372

TEST PLAN:
Run related test locally. This depends on the above compressed tensors PR.

Closed in favor of a cleaner version: https://github.com/vllm-project/llm-compressor/pull/1675

### CODE REVIEW

File: src/llmcompressor/modifiers/quantization/calibration.py [hunk 1]
```diff
@@ -124,8 +124,25 @@ def call_observer(
...
-            update_parameter_data(module, updated_zero_point, f"{base_name}_zero_point")
+            # register or update scale & zero_point parameters (supports block shapes)
+            scale_name = f"{base_name}_scale"
+            zp_name = f"{base_name}_zero_point"
+            if not hasattr(module, scale_name) or getattr(module, scale_name).shape != updated_scale.shape:
+                if hasattr(module, scale_name):
+                    delattr(module, scale_name)
+                module.register_parameter(
+                    scale_name, torch.nn.Parameter(updated_scale.clone())
+                )
```
Comment (brian-dellabetta) [inline src/llmcompressor/modifiers/quantization/calibration.py]: @kylesayrs do we need these (this and L141-143) to use `register_offload_parameter` instead?

File: src/llmcompressor/modifiers/quantization/calibration.py [hunk 2]
```diff
@@ -124,8 +124,25 @@ def call_observer(
...
                 value, g_idx=g_idx, global_scale=global_scale
             )
-            update_parameter_data(module, updated_scale, f"{base_name}_scale")
-            update_parameter_data(module, updated_zero_point, f"{base_name}_zero_point")
+            # register or update scale & zero_point parameters (supports block shapes)
+            scale_name = f"{base_name}_scale"
+            zp_name = f"{base_name}_zero_point"
+            if not hasattr(module, scale_name) or getattr(module, scale_name).shape != updated_scale.shape:
+                if hasattr(module, scale_name):
+                    delattr(module, scale_name)
```
Comment (brian-dellabetta) [inline src/llmcompressor/modifiers/quantization/calibration.py]: @kylesayrs do we need this and L140 to be `delete_offload_parameter`?

**Review by brian-dellabetta — COMMENTED** [review]
Thanks for adding this! We had some users requesting block support in 
* #1475
Couple comments for @kylesayrs , but otherwise LGTM

File: src/llmcompressor/modifiers/quantization/calibration.py [hunk 3]
```diff
@@ -124,8 +124,22 @@ def call_observer(
             updated_scale, updated_zero_point = observer(
                 value, g_idx=g_idx, global_scale=global_scale
             )
-            update_parameter_data(module, updated_scale, f"{base_name}_scale")
-            update_parameter_data(module, updated_zero_point, f"{base_name}_zero_point")
+            # register or update scale & zero_point parameters (supports block shapes)
+            scale_name = f"{base_name}_scale"
+            zp_name = f"{base_name}_zero_point"
+            for name, value in [
```
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/calibration.py:134]: This shouldn't be required? If the original name or shape doesn't match, then there's a bug upstream

File: src/llmcompressor/modifiers/quantization/calibration.py [hunk 4]
```diff
@@ -124,8 +124,22 @@ def call_observer(
...
+                (scale_name, updated_scale),
+                (zp_name, updated_zero_point),
+            ]:
+                if (
+                    not hasattr(module, name)
+                    or getattr(module, name).shape != value.shape
+                ):
+                    if hasattr(module, name):
+                        delattr(module, name)
+                    module.register_parameter(name, torch.nn.Parameter(value.clone()))
```
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/calibration.py]: This also attaches a gradient, which is not expected behavior

File: src/llmcompressor/modifiers/quantization/calibration.py [hunk 5]
```diff
@@ -124,8 +124,22 @@ def call_observer(
...
+            for name, value in [
+                (scale_name, updated_scale),
+                (zp_name, updated_zero_point),
+            ]:
+                if (
+                    not hasattr(module, name)
+                    or getattr(module, name).shape != value.shape
+                ):
+                    if hasattr(module, name):
+                        delattr(module, name)
```
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/calibration.py]: [delete_offload_parameter](https://github.com/neuralmagic/compressed-tensors/blob/180226b90b6d37ce7948c1c5db384eed65695dc9/src/compressed_tensors/utils/offload.py#L265)

File: src/llmcompressor/modifiers/quantization/calibration.py (hunk 4 above)
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/calibration.py]: [register_offload_parameter](https://github.com/neuralmagic/compressed-tensors/blob/180226b90b6d37ce7948c1c5db384eed65695dc9/src/compressed_tensors/utils/offload.py#L192)

File: src/llmcompressor/modifiers/quantization/calibration.py [hunk 6]
```diff
@@ -124,8 +124,22 @@ def call_observer(
...
+                    module.register_parameter(name, torch.nn.Parameter(value.clone()))
+                else:
+                    update_parameter_data(module, value, name)
```
Comment (kylesayrs) [inline src/llmcompressor/modifiers/quantization/calibration.py]:
```suggestion
                    update_offload_parameter(module, value, name)
```

**Review by kylesayrs — CHANGES_REQUESTED** [review]
